std/experimental/allocator/package.d

```diff
@@ -708,13 +776,18 @@ if (isInputRange!R)
             {
                 destroy(result[j]);
             }
-            alloc.deallocate(m);
+
+            static if (canSafelyDeallocPostRewind!T)
+                () @trusted { alloc.deallocate(m); } ();
+            else
+                alloc.deallocate(m);
         }
 
         for (; !range.empty; range.popFront, ++i)
         {
-            import std.conv : emplace;
-            emplace!T(result.ptr + i, range.front);
+            import std.conv : emplaceRef;
+            auto p = () @trusted { return result.ptr + i; } ();
+            emplaceRef(*p, range.front);
```

Comment:
> @wilzbach p[i] is @system

Are you sure? Isn't it the opposite way around and `p[i]` is `@safe`?

In fact `emplaceRef(p[i], range.front)` worked great for me at #4677